### DISCUSSION TITLE
Opening an SQL file stored on my IFS

### DISCUSSION THREAD

**Discussion by Gregory1701** [codefori/vscode-ibmi#2331] [Q&A]
Why in DB2fori, when I go to open a file, does it not show what is truly in my IFS 'SQL Files' folder?
![image](https://github.com/user-attachments/assets/c9bdeb18-42d5-443f-b620-a31f2fb92acf)

<hr />

|Context|Version|
|-|-|
|Code for IBM i version|2.13.6|
|Visual Studio Code version|1.95.1|
|Operating System|linux_x64|

<details>
<summary>Active extensions</summary>

```
Code for IBM i User Spooled Files (vscode-ibmi-UserSpooledFiles): 0.1.1
Code for IBM i Walkthroughs (vscode-ibmi-walkthroughs): 0.5.0
Configuration Editing (configuration-editing): 1.0.0
Db2 for IBM i (vscode-db2i): 1.6.1
Emmet (emmet): 1.0.0
Extension Authoring (extension-editing): 1.0.0
Git (git): 1.0.0
Git Base (git-base): 1.0.0
GitHub (github): 0.0.1
GitHub Copilot (copilot): 1.243.0
GitHub Copilot Chat (copilot-chat): 0.22.1
Hawkeye Pathfinder (vscode-ibmi-hawkeye): 0.0.13
IBM Granite.Code (wca-core): 1.0.0
IBM i Debug (ibmidebug): 2.0.1
JSON Language Features (json-language-features): 1.0.0
Jira and Bitbucket (Atlassian Labs) (atlascode): 3.0.10
Local Tunnel Port Forwarding (tunnel-forwarding): 1.0.0
Merge Conflict (merge-conflict): 1.0.0
NPM support for VS Code (npm): 1.0.1
Node Debug Auto-attach (debug-auto-launch): 1.0.0
Postman (postman-for-vscode): 1.4.1
RPGLE (vscode-rpgle): 0.26.12
Server Ready Action (debug-server-ready): 1.0.0
TypeScript and JavaScript Language Features (typescript-language-features): 1.0.0
YAML (vscode-yaml): 1.15.0
```
</details>

<hr />

<details>
<summary>Remote system</summary>

|Setting|Value|
|-|-|
|IBM i OS|V7R5M0|
|Tech Refresh|n/a|
|CCSID Origin|37|
|Runtime CCSID|37|
|Default CCSID|37|
|SQL|Enabled
|Source dates|Disabled

### Enabled features

|/QOpenSys/pkgs/bin|/usr/bin|/QSYS.lib/ILEDITOR.lib|/QSYS.LIB|/QIBM/ProdData/IBMiDebugService/bin|/QOpenSys/QIBM/ProdData/JavaVM/jdk80|/QOpenSys/QIBM/ProdData/JavaVM/jdk11|/QOpenSys/QIBM/ProdData/JavaVM/jdk17|
|-|-|-|-|-|-|-|-|
|bash|attr|GETNEWLIBL.PGM|QZDFMDB2.PGM|startDebugService.sh|64bit|64bit|64bit|
|find|iconv|||||||
|git|setccsid|||||||
|grep||||||||
|ls||||||||
|md5sum||||||||
|sort||||||||
|stat||||||||
|tar||||||||
|tn5250||||||||
</details>

<details>
<summary>Shell env</summary>

```bash
```
</details>

<details>
<summary>Variants</summary>

```json
{
  "american": "#@$",
  "local": "#@$"
}
```
</details>

<details>
<summary>Errors</summary>

```json
[
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPF9801:  Object VSC00AFN87 in library ILEDITOR not found.",
    "cwd": "/home/PGMGREGORY"
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "CPD0043:  Keyword SQL not valid for this command.\nCPD0043:  Keyword COMMIT not valid for this command.\nCPD0043:  Keyword NAMING not valid for this command.\nCPF0001:  Error found on RUNSQL command.",
    "cwd": "/"
  },
  {
  
]
```
</details>

**Comment by worksofliam** [comment]
The 'Open' you're using isn't really made for navigating the IFS. I do recommend you simply using the IFS Browser to navigate to your sources for the time being.

**Reply by Gregory1701** [reply]
Ah, ok. I created a filter in my IFS browser and can open my saved SQL files that way, which works great.
![image](https://github.com/user-attachments/assets/4aa938a2-ce24-45fb-9005-7496885a726c)
